Add Warmup Operation in Perf-tool Benchmarking Tests

naveentatikonda opened the pull request:
### Description
Add Warmup Operation in Perf-tool Benchmarking Tests.
 
### Check List
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

naveentatikonda commented:
Build failure is not related to these changes.

navneet1v commented:
@naveentatikonda can we add warmup config in all other release config like hnsw-pq, ivf, ivf-pq and filters for ivf

martin-gaievski commented on benchmarks/perf-tool/okpt/test/steps/steps.py:
```diff
@@ -690,6 +709,23 @@ def delete_model(endpoint, port, model_id):
+    Returns:
+        number of shards the plugin succeeded and failed to warm up.
+    """
+    response = requests.get('http://' + endpoint + ':' + str(port) +
```
Seems we're using default timeout here, what is the chance that request will take longer than that time?

naveentatikonda replied:
I think our warmup API doesn't accept a timeout parameter. So, I think we can add a static sleep time of 5 min. WDYT ?

martin-gaievski replied:
that should work. We can do it as a parameter for this step, and use 5 mins as a default value

naveentatikonda replied:
As discussed offline, made the timeout parameter configurable to fix the issue

naveentatikonda commented:
> @naveentatikonda can we add warmup config in all other release config like hnsw-pq, ivf, ivf-pq and filters for ivf

I have already added it in all release config. I missed a couple of them in sample config. I have added it now.

navneet1v commented:
> > @naveentatikonda can we add warmup config in all other release config like hnsw-pq, ivf, ivf-pq and filters for ivf
> 
> I have already added it in all release config. I missed a couple of them in sample config. I have added it now.

You missed on the filtering configs for the algorithms

naveentatikonda commented:
> > > @naveentatikonda can we add warmup config in all other release config like hnsw-pq, ivf, ivf-pq and filters for ivf
> > 
> > 
> > I have already added it in all release config. I missed a couple of them in sample config. I have added it now.
> 
> You missed on the filtering configs for the algorithms

Oh sorry, Thanks for catching it. Updated them.
